src/core/shape_inference/include/utils.hpp

```diff
@@ -44,37 +47,145 @@ void eltwise_shape_infer(const OpType* op, const std::vector<T>& input_shapes, s
+
+    switch (et) {
+    case element::Type_t::i4: {
+        using dtype = fundamental_type_for<element::Type_t::i4>;
+        std::transform(static_cast<const dtype*>(ptr), static_cast<const dtype*>(ptr) + size, out_it, func);
+    } break;
+    case element::Type_t::i8: {
+        using dtype = fundamental_type_for<element::Type_t::i8>;
+        std::transform(static_cast<const dtype*>(ptr), static_cast<const dtype*>(ptr) + size, out_it, func);
+    } break;
+    case element::Type_t::i16: {
+        using dtype = fundamental_type_for<element::Type_t::i16>;
+        std::transform(static_cast<const dtype*>(ptr), static_cast<const dtype*>(ptr) + size, out_it, func);
+    } break;
+    case element::Type_t::i32: {
+        using dtype = fundamental_type_for<element::Type_t::i32>;
+        std::transform(static_cast<const dtype*>(ptr), static_cast<const dtype*>(ptr) + size, out_it, func);
+    } break;
+    case element::Type_t::i64: {
+        using dtype = fundamental_type_for<element::Type_t::i64>;
+        std::transform(static_cast<const dtype*>(ptr), static_cast<const dtype*>(ptr) + size, out_it, func);
+    } break;
+    case element::Type_t::u4: {
+        using dtype = fundamental_type_for<element::Type_t::u4>;
+        std::transform(static_cast<const dtype*>(ptr), static_cast<const dtype*>(ptr) + size, out_it, func);
+    } break;
+    case element::Type_t::u8: {
+        using dtype = fundamental_type_for<element::Type_t::u8>;
+        std::transform(static_cast<const dtype*>(ptr), static_cast<const dtype*>(ptr) + size, out_it, func);
+    } break;
+    case element::Type_t::u16: {
+        using dtype = fundamental_type_for<element::Type_t::u16>;
+        std::transform(static_cast<const dtype*>(ptr), static_cast<const dtype*>(ptr) + size, out_it, func);
+    } break;
+    case element::Type_t::u32: {
+        using dtype = fundamental_type_for<element::Type_t::u32>;
+        std::transform(static_cast<const dtype*>(ptr), static_cast<const dtype*>(ptr) + size, out_it, func);
+    } break;
+    case element::Type_t::u64: {
+        using dtype = fundamental_type_for<element::Type_t::u64>;
+        std::transform(static_cast<const dtype*>(ptr), static_cast<const dtype*>(ptr) + size, out_it, func);
+    } break;
+    default:
+        OPENVINO_ASSERT(false, "Not supported element type ", et);
+    };
+    return out;
+}
+
+/**
+ * \brief Get data from Host tensor as object TResult.
+ *
+ * \tparam T               TResult data type.
+ * \tparam TResult         Type of return object, must support creation of std::inserter. Default std::vector<T>.
+ * \tparam UnaryOperation  Unary function object applied on data with signature (T f(const U u)).
+ *
+ * \param tv    Input host tensor.
+ * \param func  Unary operation function object.
+ *
+ * \return Object of TResult with data from host tensor.
+ */
+template <class T, class TResult = std::vector<T>, class UnaryOperation>
+TResult get_tensor_data_as(HostTensor& tv, UnaryOperation func) {
```

Comment:
std::forward the functor here (and other locations below)

Reply:
Done